Source/Core/DolphinQt2/Settings/AudioPane.cpp

```diff
@@ -0,0 +1,227 @@
+#include <QComboBox>
+#include <QFormLayout>
+#include <QGridLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QRadioButton>
+#include <QSlider>
+#include <QSpinBox>
+#include <QVBoxLayout>
+
+#include "AudioCommon/AudioCommon.h"
+#include "DolphinQt2/Config/SettingsWindow.h"
+#include "DolphinQt2/Settings.h"
+
+AudioPane::AudioPane()
+{
+  CreateWidgets();
+  LoadSettings();
+  ConnectWidgets();
+
+  connect(&Settings::Instance(), &Settings::VolumeChanged, this, &AudioPane::OnVolumeChanged);
+}
+
+void AudioPane::CreateWidgets()
+{
+  auto* dsp_box = new QGroupBox(tr("DSP Emulator Engine"));
+  auto* dsp_layout = new QVBoxLayout;
+  dsp_box->setLayout(dsp_layout);
+  m_dsp_hle = new QRadioButton(tr("DSP HLE emulation (fast)"));
+  m_dsp_lle = new QRadioButton(tr("DSP LLE recompiler"));
+  m_dsp_interpreter = new QRadioButton(tr("DSP LLE interpreter (slow)"));
+
+  dsp_layout->addWidget(m_dsp_hle);
+  dsp_layout->addWidget(m_dsp_lle);
+  dsp_layout->addWidget(m_dsp_interpreter);
+
+  auto* volume_box = new QGroupBox(tr("Volume"));
+  auto* volume_layout = new QGridLayout;
+  m_volume_slider = new QSlider;
+  m_volume_indicator = new QLabel();
+
+  volume_box->setLayout(volume_layout);
+
+  m_volume_slider->setMinimum(0);
+  m_volume_slider->setMaximum(100);
+
+  m_volume_indicator->setAlignment(Qt::AlignVCenter | Qt::AlignHCenter);
+
+  volume_layout->addWidget(m_volume_slider, 0, 1);
+  volume_layout->addWidget(m_volume_indicator, 1, 0, 1, 3);
+
+  auto* backend_box = new QGroupBox(tr("Backend Settings"));
+  auto* backend_layout = new QFormLayout;
+  backend_box->setLayout(backend_layout);
+  m_backend_label = new QLabel(tr("Audio Backend"));
+  m_backend_combo = new QComboBox();
+  m_latency_label = new QLabel(tr("Latency"));
+  m_dolby_pro_logic = new QCheckBox(tr("Dolby Pro Logic II decoder"));
+  m_latency_spin = new QSpinBox();
+
+  m_latency_spin->setMinimum(0);
+  m_latency_spin->setMaximum(30);
+
+  backend_layout->addRow(m_backend_label, m_backend_combo);
+  backend_layout->addRow(m_latency_label, m_latency_spin);
+  backend_layout->addRow(m_dolby_pro_logic);
+
+  auto* stretching_box = new QGroupBox(tr("Audio Stretching Settings"));
+  auto* stretching_layout = new QGridLayout;
+  m_stretching_enable = new QCheckBox(tr("Enable Audio Stretching"));
+  m_stretching_buffer_slider = new QSlider(Qt::Horizontal);
+  m_stretching_buffer_indicator = new QLabel();
+  m_stretching_buffer_label = new QLabel(tr("Buffer Size"));
+  stretching_box->setLayout(stretching_layout);
+
+  m_stretching_buffer_slider->setMinimum(5);
+  m_stretching_buffer_slider->setMaximum(300);
+
+  stretching_layout->addWidget(m_stretching_enable, 0, 0, 1, -1);
+  stretching_layout->addWidget(m_stretching_buffer_label, 1, 0);
+  stretching_layout->addWidget(m_stretching_buffer_slider, 1, 1);
+  stretching_layout->addWidget(m_stretching_buffer_indicator, 1, 2);
+
+  m_main_layout = new QGridLayout;
+
+  m_main_layout->addWidget(dsp_box, 0, 0, 1, 4);
```

Comment:
Because that seems like a sane rate and I can't figure out another way to do it?